### DISCUSSION TITLE
Allow control over the default "Relate to original" behavior

### DISCUSSION THREAD

**Discussion by mennomout** [umbraco/Umbraco-CMS#13802] [Features and ideas]
Currently whenever you copy a content node, the toggle for "Relate to original" is set to `true` by default. However often this is not the desired default. It would be a quality of life improvement if we'd be able to manage it's default behavior. This issue was brought up before on OurUmbraco: https://our.umbraco.com/forum/using-umbraco-and-getting-started/98295-relate-to-original

Besides managing the default for this toggle, it wouldn't hurt to make the "Include descendants" toggle default manageable as well.

**Comment by bergmania** [comment]
Sounds like a good idea. I will set this up-for-graps. We need to ensure this is also implemented for the new backoffice (or only in that version).

**Comment by tormnator** [comment]
I think the defaults should be set to false, and then the extra feature would be to make this a configurable setting. This way if you forget to look at those two checkboxes it's a less severe error. Also keep in mind that on smaller screens the checkboxes aren't visible unless you make the effort to scroll down and look for them (assuming you now they exist). Thinking about it, an improvement would be to put them on a "sticky" (always visible) panel, or maybe move them to the top.

**Comment by c9mb** [comment]
I agree with "false" as default, as well as a way to configure/manage the defaults. Although I understand how related copies can be useful, I don't think that I've ever wanted it in all the times I've used it.

**Comment by mennomout** [comment]
@nul800sebastiaan There was some discussion on the Discord about this issue that I would like to continue here. Before I continue with the development of this feature it's important to have consensus about the solution. There are clear cases when you always want to have this feature on (current default behavior) and where having it off by default could lead to major problems, specifically on multilingual sites. My question is, would making it configurable in the appsettings be a good solution to this issue? Leaving the default behavior as is (true by default) but allowing developers to change the behavior to fit the needs of a specific customer?

**Comment by KevinJump** [comment]
Hello, awkward squad here 🖐️ 

Relate to orginal's 'orginal' purpose was and is to relate the content from one tree to another, principally when people are running multi-lingual setups within umbraco (not multi-variant ones, but ones where they have multiple trees one for each language).

in these senarios relate to orginal keeps a link between the original content and the same content in another language.  (Often You can't just work this out at run time because about us in english may well have a different node name when its about us in french).

This then allows you to do two key thing (probibly more) :

1) work out the link from one language version of the page to the same page in other languages - so you can build something on your site to flick between languages and keep people on the same page (e.g a dlanguage drop down in the menu).

2) It also allows for translations to be send to the right location when someone wants to translate the english version of the page to french the translation tool knows where the french page actually is. 

In my experience both of these features are built/used by people way after they have created the individual language trees. Indeed we see people asking "how do we get this translation into the right place" - because they unticked the relate to original box when they first did it. 

If you copy without relate to original then at a later date go , oh we needed that, you have to re-link your entire site, by hand which is a slow and very laborious task, as you have to link each page*  

So having this as default does save a lot of people a lot of time as it gives them a feature they didn't realize they needed.

-- 

That said, i think calling this feature 'relate to original' is probably the cause of the problem - it doesn't really make sense for what it does. It's more of a "link languages" or "copy as language" feature (both those names suck too).

Evidence from talking to people who have / haven't set this - is they think it does some kind of content magic when it 'relates' the original to the copy - and so they either untick it or think it is giving them something its not. 


_* Linking every page really can be a pain, we have a package to help people do it, but it can take ages. Also In Translation manager people can choose to retranslate the site without the links and translation manager will build the links as it goes, but it will also create duplicate content because it doesn't know for certain that page a is page b on another site._

**Comment by nathanwoulfe** [comment]
My 2c, having never built multi-lingual sites... 

Based on Kevin's informed insights, I'd suggest if this were made configurable it might make better sense to configure whether the control is displayed at all - that way, the relations are still generated but users don't need to consider changing the value. Maybe it makes sense to provide configuration for the default value and display (which is the approach we've taken in some of the commercial packages where there's a lot of configuration options, but not all of those always makes sense to surface/enable in the Backoffice.

Changing the default to false is a behavioural breaking change, even if it is configurable, so really this discussion should be focussing on whether the default should be configurable, not whether it should be changed.

**Comment by tormnator** [comment]
Our need for "Relate to Original" is as @KevinJump describes, to relate "identical" pages in different languages (between language specific subsites). After creating a new page on one subsite you copy it to the other subsite(s) and then translate the contents. This is when you want to keep the relation. In almost all other cases, e.g. you're copying a page for testing or to create a similar page in the same subsite, you want to turn off the relation.

I'd say in our case, the person copying from one language site to another knows what they're doing and will ensure the checkbox is correctly ticked "on". The problem is when an editor is working on and is copying content pages within a subsite, they don't necessarily understand what "Relate to Original" means, and they probably shouldn't have to worry about it.

The main thing, in my mind, is to make those two checkboxes visible (move to top, or make them sticky), and to explain what their purpose is.

**Comment by jgettrup-ng** [comment]
I'd like to add a related usability concern that we've encountered with editors.

The bigger issue for us isn't the default value of Include descendants, it's that the option is easy to miss entirely.

In the Duplicate To dialog, the destination tree is displayed before the options. On sites with larger content trees, the tree can occupy most or all of the visible area of the dialog, pushing the options below the fold. As a result, many editors never see the Include descendants checkbox and proceed with the copy operation using the defaults.

We regularly see users accidentally copy only the selected node when they intended to copy the entire subtree. They then have to delete the copied content and repeat the operation.

I think this could be improved by:

Moving the options section above the tree picker, or
Keeping the options section visible/sticky while navigating the tree.

From a workflow perspective, it feels more natural to:

Configure the copy operation.
Choose the destination.
Execute the action.

This would make the available options much more discoverable and reduce user error, especially on installations with deep or extensive content structures.

For context, we're a large organisation with a substantial content tree, and this is a recurring issue for our content editors.
